### DISCUSSION TITLE
How to use an Automation Plugin in and Automation?

### DISCUSSION THREAD

**Discussion by BrianLakstins** [Budibase/budibase#11199] [Help]
I see that an Automation Plugin can do just about anything.  It looks like it runs as a Node app that can use other NPM packages.

How does it get used in an automation?  I have not created an Automation Plugin yet, but I when I watched the videos and looked through the documentation, I did not see where selecting an Automation Plugin was part of the Automation flow.

Do Automation Plugins only show up as part of the Automation process if I have an Automation plugin uploaded?  Are they run by using Javascript and referencing them in some way?

**Comment by ConorWebb96** [comment]
Hey @BrianLakstins,

Yes you are right, the new automation plugin will only appear whenever the plugin has been installed (If you are using the cli you will need to install it, then run a few commands e.g. `npm i`, `npm run watch` within its directory). Its purely up to how you make them, the base files use Typescript to code them. Additional information on this can be found here; https://docs.budibase.com/docs/custom-automations#automationts

From what I understand of the plugin. This doesn't add any triggers, it allows you to add custom actions. E.g. steps that you can make specifically for your automation's needs. An example of this could be doing some complex queries with a check that it needs to pass to continue with the automation. It really depends on your use case, it does give you a lot of power of that you can add. Additionally, you can add multiple steps meaning you can create a completely custom automation filled with your custom actions.

I'd recommend also reading up on the schema and its properties and types as it aids in understand just how much you can do with this plugin. https://docs.budibase.com/docs/automation-schema#schema-structure

I hope this helps.
